src/__mocks__/firebase/app.js

```diff
@@ -1,16 +1,26 @@
 var userChangedCb = null;
 
 export const signInCalled = jest.fn();
+import { validEmail, validPassword, Invalid } from '../../constants/testData';
 
 const authObject = {
   signInWithEmailAndPassword: async (email, pw) => {
     signInCalled();
-    if (email && pw)
+    if (email === validEmail && pw === validPassword)
       return Promise.resolve('Hello');
     else {
-      const error = {
-        code: 'auth/invalid-email'
+      var error = {
+        code: 'auth/unknown'
       };
+      if (email === validEmail && pw === Invalid) {
```

Comment:
start all variable names with lower case letter and add suffix to note what is invalid. For example invalidPassword and at emal invalidEmail